ClemBot.Bot/bot/cogs/trivia_cog.py

```diff
@@ -0,0 +1,566 @@
+#Stephen Martin
+#5/9/2022
+#trivia_cog.py
+
+import json
+import logging
+import uuid
+import random
+import aiohttp
+import html
+import asyncio
+import typing as t
+from dataclasses import dataclass
+import discord
+import discord.ext.commands as commands
+from discord.ext.commands.errors import UserInputError
+import bot.bot_secrets as bot_secrets
+import bot.extensions as ext
+from bot.consts import Colors
+from bot.messaging.events import Events
+
+log = logging.getLogger(__name__)
+
+
+class TriviaCog(commands.Cog):
+
+    def cog_unload(self):
+
+        self.session.close()
+
+    def __init__(self, bot):
+        self.messages = {}
+        self.bot = bot
+        self.session = aiohttp.ClientSession(headers={'Connection': 'keep-alive'})
+
+    @ext.group(case_insensitive=True, invoke_without_command=True)
+    @ext.long_help(
+        "Use !trivia to return a random assortment of 10 trivia questions. React with emojis to submit your answer choice")
+    @commands.cooldown(1, 25, commands.BucketType.user)
+    @ext.short_help(
+        "!trivia use !trivia m for custom arguments")
+    async def trivia(self, ctx):
+
+        async with await self.session.get(DEFAULT_URL) as resp:
+            parse_text = await resp.text()
+            new_response = json.loads(parse_text)
+            parsed_response = await self.html_parser(new_response)
+
+            #If you're curious as to why this doesn't check response code: It's because it will never NOT have questions for the default. If it does the website is down and it will error out anyway.
+
+        best_list = await self.dict_publisher(parsed_response)  #Publishes the dictionary
+
+        new_task = await self.asyncio_publisher(ctx, best_list[1])  #This returns key values for our list
+
+        the_reaction = asyncio.create_task(
+            self.on_reaction(ctx, new_task[3]))  #Starts the event listener for the reaction BEFORE emojis are sent
+
+        task1 = asyncio.create_task(self.send_scroll_reactions(new_task[0], new_task[1], new_task[2]))  #Sends the answer key
+
+        user_reaction = await the_reaction  #Gets the user's reaction
+
+        await self.parse_reaction(new_task[0], user_reaction[0], user_reaction[1], best_list[0])  #Parses the user reaction
+
+        while not task1.done():  #Loops reading the user's
+
+            new_reaction = await self.on_reaction(ctx, new_task[3])
+            await self.parse_reaction(new_task[0], new_reaction[0], new_reaction[1], best_list[0])
+
+        return
+
+    @trivia.command(aliases=['m'])
+    @ext.long_help(
+        "Specify arguments you want to return such as question number (max 35), category, difficulty, or question type. Use numbers for quicker specification of category by typing in the number beside the category in !help. Use 0 for unused categories!")
+    @ext.short_help("!trivia m <Question Number> <category/substring/number> <Difficulty/Number> <question type>")
```

Comment:
the short help is just a shorter version of long help to show in the help embed home page, this stuff should be in the example section